### PR TITLE
net: wifi: Current max rate and over run statistics

### PR DESCRIPTION
**Pull request by kapbh**
Current PHY rate
It represents the current PHY rate of transfer of data in bits per second. It will a TX data rate.

Over run count
It represents the number of packets dropped either at received and sent due to lack of buffer memory to retain all packets on the network interface.

### CODE REVIEW

File: include/zephyr/net/net_stats.h [hunk 1]
```diff
@@ -623,6 +623,12 @@ struct net_stats_wifi {
 
 	/** Total number of unicast packets received and sent */
 	struct net_stats_pkts unicast;
+
+	/** The current maximum PHY rate of transfer of data*/
+	net_stats_t current_max_rate;
```
Comment (jukkar) [inline include/zephyr/net/net_stats.h:628]: This does not feel very statistics like variable. Would using #76062 give better real time information for the rate?

Comment (krish2718) [reply include/zephyr/net/net_stats.h:628]: `wifi_iface_status` is a better place for this, it's mostly depends on `enum wifi_link_mode link_mode;` and capabilities of the peer.

File: include/zephyr/net/net_stats.h [hunk 2]
```diff
@@ -623,6 +623,9 @@ struct net_stats_wifi {
 
 	/** Total number of unicast packets received and sent */
 	struct net_stats_pkts unicast;
+
+	/** Total number of dropped packets at received and sent*/
+	net_stats_t overrun_count;
```
Comment (krish2718) [inline include/zephyr/net/net_stats.h:628]: please split this into two commits.

File: include/zephyr/net/wifi_mgmt.h [hunk 3]
```diff
@@ -522,6 +522,8 @@ struct wifi_iface_status {
 	unsigned short beacon_interval;
 	/** is TWT capable? */
 	bool twt_capable;
+	/** The current maximum PHY rate of transfer of data */
+	int current_max_rate;
```
Comment (krish2718) [inline include/zephyr/net/wifi_mgmt.h:526]: `current_max_phy_rate`

File: subsys/net/l2/wifi/wifi_shell.c [hunk 4]
```diff
@@ -877,6 +878,7 @@ static void print_wifi_stats(struct net_if *iface, struct net_stats_wifi *data,
 	PR("Beacons missed   : %u\n", data->sta_mgmt.beacons_miss);
 	PR("Unicast received : %u\n", data->unicast.rx);
 	PR("Unicast sent     : %u\n", data->unicast.tx);
+	PR("Over run count   : %u\n", data->overrun_count);
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:881]:
```suggestion
	PR("Overrun count   : %u\n", data->overrun_count);
```

File: include/zephyr/net/wifi_mgmt.h [hunk 5]
```diff
@@ -522,6 +522,8 @@ struct wifi_iface_status {
 	unsigned short beacon_interval;
 	/** is TWT capable? */
 	bool twt_capable;
+	/** The current maximum PHY rate of transfer of data */
```
Comment (krish2718) [inline include/zephyr/net/wifi_mgmt.h:525]:
```suggestion
	/** The current maximum 802.11 PHY data rate */
```

**Review by krish2718 — DISMISSED** [review]

**Review by krish2718 — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by jukkar — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**